staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -300,3 +307,61 @@ func (le *LeaderElector) maybeReportTransition() {
 		go le.config.Callbacks.OnNewLeader(le.reportedLeader)
 	}
 }
+
+type LeaderChecker interface {
+	healthz.HealthzChecker
+	SetLeaderElection(le *LeaderElector)
+}
+
+type timeNow func() time.Time
+
+type leaderChecker struct {
+	c           *apiserverconfig.LeaderElectionConfiguration
+	pointerLock sync.Mutex
+	le          *LeaderElector
+	timeNow
+}
+
+func (l *leaderChecker) Name() string {
+	return "leaderElection/" + l.c.ResourceLock
+}
+
+func (l *leaderChecker) Check(req *http.Request) error {
+	if !l.c.LeaderElect {
```

Comment:
Perhaps don't construct the leaderChecker in the first place if this is false?

Reply:
Done